SafeShutdown.py

```diff
@@ -16,11 +16,13 @@
 #functions that handle button events
 def when_pressed():
   led.blink(.2,.2)
-  os.system("sudo killall emulationstation && sleep 5s && sudo shutdown -h now")
+  os.system("sudo killall emulationstation")
```

Comment:
@AGobron 

Hi,

If you use ```killall -w emulationstation```, ```killall``` will wait for the process to finish, before continuing (blocking mode), you can then remove the 5 seconds sleep below this line.

Reply:
This seems like a bad idea.  From the man page: "Note that killall may wait forever if the signal was ignored, had no effect, or if the process stays in zombie state."

Reply:
@madzarevic 

Never had any issues with ```killall -w```, we use it extensively in DietPi.

If the process endless hangs ```killall -w``` then we can assume the system is unstable and should be resolved correctly, or, ```emulationstation``` has more in-depth issues.

Reply:
@Fourdee I've never had any issues with killall -w either.  The question is what would you want to happen *if* there is a problem.  Maybe you might want the shutdown to abort to allow further investigation.  I personally would rather the pi just continue to shutdown so I don't need to go find a keyboard or get an ssh session going just to get the pi to safely shutdown so I don't need to pull the power plug and risk corrupting the SD card.  Presumably there is a good chance I can still investigate any issues at a later time if I really want to, which is made much easier if the filesystem is still intact.

Reply:
FYI, per my understanding, the key perk of the 5-second delay is so emulationstation can write out metadata/config changes...  Not sure how long it takes now, but it used to take forever.